File: src/main/java/com/github/morningzeng/toolset/enums/DataToBinaryTypeEnum.java
package com.github.morningzeng.toolset.enums;

import lombok.SneakyThrows;
import org.apache.commons.codec.binary.Hex;

import java.nio.charset.StandardCharsets;
import java.util.Base64;

/**
 * @author Morning Zeng
 * @since 2024-05-23
 */
public enum DataToBinaryTypeEnum {

    TEXT {
        @Override
        public byte[] bytes(final String data) {
            return data.getBytes(StandardCharsets.UTF_8);
        }
    },
    HEX {
        @SneakyThrows
        @Override
        public byte[] bytes(final String data) {
            return Hex.decodeHex(data);
        }
    },
    BASE64 {
        @Override
        public byte[] bytes(final String data) {
            return Base64.getDecoder().decode(data);
        }
    },
    ;

    public abstract byte[] bytes(final String data);

}

File: src/main/kotlin/com/github/morningzeng/toolbox/enums/DataToBinaryTypeEnum.kt
package com.github.morningzeng.toolbox.enums

import org.apache.commons.codec.binary.Hex
import java.nio.charset.StandardCharsets
import java.util.*

/**
 * @author Morning Zeng
 * @since 2025-05-20
 */
enum class DataToBinaryTypeEnum {

    BASE64 {
        override fun bytes(data: String): ByteArray = Base64.getDecoder().decode(data)

        override fun toString(bytes: ByteArray): String = Base64.getEncoder().encodeToString(bytes)
    },
    HEX {
        override fun bytes(data: String): ByteArray = Hex.decodeHex(data)

        override fun toString(bytes: ByteArray): String = Hex.encodeHexString(bytes)
    },
    TEXT {
        override fun bytes(data: String): ByteArray = data.toByteArray(StandardCharsets.UTF_8)

        override fun toString(bytes: ByteArray): String = String(bytes, StandardCharsets.UTF_8)
    },


    ;

    abstract fun bytes(data: String): ByteArray

    abstract fun toString(bytes: ByteArray): String

}